### PR TITLE
Silence all println statements by default; add boolean switch to enable

### LINKED ISSUE #22: Remove println statements in favour of logging framework
I'm attempting to use this extension in a command line application I'm working on.  The println statements throughout are polluting my console window at runtime with no way to suppress.  Would you be willing to replace those println statements with a logging framework (log4j, slf4j, etc.)?  I'll do a pull request if you're willing to merge it, I just don't want to go to the trouble if you'll never accept a pull request that adds a dependency.

Alternatively, a config option or system property to suppress the println statements would be the next best option.  A logging framework means other users can more easily control what output goes where.

### PR DESCRIPTION
**Pull request by Slugger**
Fixes #22

### CODE REVIEW

File: src/main/groovy/com/athaydes/spockframework/report/SpockReportExtension.groovy [hunk 1]
```diff
@@ -40,31 +41,41 @@ class SpockReportExtension implements IGlobalExtension {
...
+				println "Error configuring ${this.class.name}! $e"
+			}
 		}
 	}
 
 	def instantiateReportCreator() {
 		def reportCreatorClass = Class.forName( reportCreatorClassName )
-		reportCreatorClass.asSubclass( IReportCreator ).newInstance()
+		def obj = reportCreatorClass.asSubclass( IReportCreator ).newInstance()
+		obj.silenceOutput = silenceOutput
```
Comment (renatoathaydes) [inline src/main/groovy/com/athaydes/spockframework/report/SpockReportExtension.groovy]: This is not correct. The reportCreator is a subtype of `IReportCreator` but is not necessarily an `AbstractHtmlCreator` (my plan is to add other report creators later), even if currently that's the case.

**Comment by renatoathaydes** [comment]
I don't like adding a boolean to every class in the project just to be able to silence output. A more resilient approach would've been to create a really simple Logger, just like a "real" logger framework, but that can be configured via the spock-reports config file.... and let people choose whether they want to log at level "INFO" or "WARN" (for example). While I think using a real logger is a bad idea due to dependency issues, I do think the principles of logging frameworks should be followed.

I was even thinking of using Groovy dynamism to do this... something like using `@Category` to mixin a `spockframework.report.Logger` to each class, then just do `log("blah")` which will "magically" be forwarded to the `Logger` Category.... which can have access to the config to decide whether to print anything or not...

**Comment by renatoathaydes** [comment]
I was thinking a little more about this!! Why don't we just use Groovy's `@Log` annotation?
That only requires `java.util.logging` so people can just use the plain java logging config mechanism to decide what they want spock-reports to print out....?

http://groovy.codehaus.org/Logging

**Comment by Slugger** [comment]
`@Log` makes my life a lot easier. :)  Wasn't sure you wanted any dependency at all added otherwise I'd have went that way from the start.

I'm going to replace it all with the `@Log`, but I'll leave any additionally logging you want to add to you?

I'm going to do this as a new pull request from a new branch.

**Pull request closed without merging**